api/base/parsers.py

```diff
@@ -25,14 +50,31 @@ def parse(self, stream, media_type=None, parser_context=None):
         data = result.get('data', {})
 
         if data:
-            if 'attributes' not in data:
-                raise JSONAPIException(source={'pointer': '/data/attributes'}, detail=NO_ATTRIBUTES_ERROR)
+            relationships = data.get('relationships')
+
+            path = stream.path
+            related_resources = ['contributors', 'node_links']
```

Comment:
I don't think you have access to the serializer class here.  We're using the parser to validate a lot of the JSONAPI request format.  There are some things that are always going to be required in the request. `data`, for example, or `type`.  But `relationships` are needed in the request only when creating node contributors or node links, currently. 

I was using the `related resources` to determine if the parser needed to make sure either `attributes` or `relationships` were included in the request.